src/cpp/rtps/transport/TCPChannelResourceSecure.cpp

```diff
@@ -339,6 +353,7 @@ void TCPChannelResourceSecure::close()
 void TCPChannelResourceSecure::shutdown(
         asio::socket_base::shutdown_type)
 {
+    // WARNING: This function blocks until receiving the peer’s close_notify
```

Comment:
After some reading I found this possibility, however I am not considering dealing with it in this PR.